test/state/bloom_filter.cpp

```diff
@@ -0,0 +1,58 @@
+// evmone: Fast Ethereum Virtual Machine implementation
+// Copyright 2023 The evmone Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "bloom_filter.hpp"
+#include "state.hpp"
+#include <intx/intx.hpp>
+
+namespace evmone::state
+{
+
+namespace
+{
+/// Adds an entry to the bloom filter
+/// based on
+/// https://ethereum.github.io/execution-specs/autoapi/ethereum/shanghai/bloom/index.html#add-to-bloom
+inline void add_to(BloomFilter& bf, const bytes_view& entry)
+{
+    const auto hash = keccak256(entry);
+
+    // take the least significant 11-bits of the first three 16-bit values
+    for (const auto i : {0, 2, 4})
+    {
+        const auto bit_to_set = ((hash.bytes[i] & 0x07) << 8) + hash.bytes[i + 1];
+        const auto bit_index = 0x07FF - bit_to_set;
+        const auto byte_index = static_cast<uint8_t>(bit_index / 8);
```

Comment:
```suggestion
        const auto byte_index = bit_index / 8;
```
This cast is likely not needed but you can want to use `size_t` type.